InvalidJavadocPosition false-positive for record compact constructor with package-private accessibility

I have read check documentation: https://checkstyle.org/checks/javadoc/invalidjavadocposition.html#InvalidJavadocPosition
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
javac com\company\Testcase.java
```

```
java -jar checkstyle-10.25.0-all.jar -c /sun_checks.xml com\company\Testcase.java
Starting audit...
[ERROR] C:\Users\Gili\Documents\docker\core\src\main\java\com\company\Testcase.java:14:5: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
Audit done.
Checkstyle ends with 1 errors.
```

```
java -jar checkstyle-10.25.0-all.jar -c /google_checks.xml com\company\Testcase.java
Starting audit...
[WARN] C:\Users\Gili\Documents\docker\core\src\main\java\com\company\Testcase.java:14:5: Javadoc comment is placed in the wrong location. [InvalidJavadocPosition]
Audit done.
```

```bash
type com\company\Testcase.java
package com.company;

/**
 * Some description.
 */
public class Testcase {
  /**
   * The configuration of a bind mount.
   *
   * @param containerPath a path on the container
   * @param options       mounting options
   */
  record BindMount(String containerPath, String... options) {
    /**
     * Creates a mount.
     *
     * @param containerPath a path on the container
     * @param options       mounting options
     * @throws NullPointerException     if any of the arguments are null
     */
    BindMount {
    }
  }
}
```

---

**Describe what you expect in detail.**
No warnings or errors.

---

PS: If you use a canonical constructor instead of a compact constructor, the error goes away. Similarly, if you use `public` accessibility instead of package-protected, the error goes away.

https://github.com/checkstyle/checkstyle/blob/b22b1b42e8ecc9f87b0a9fcd228157cdf2de1516/src/main/resources/google_checks.xml#L365

--- a/config/google-java-format/excluded/compilable-input-paths.txt
+++ b/config/google-java-format/excluded/compilable-input-paths.txt
@@ -63,6 +63,10 @@ src/it/resources/com/google/checkstyle/test/chapter4formatting/rule489textblocks
 src/it/resources/com/google/checkstyle/test/chapter5naming/rule522classnames/InputClassNames.java
 src/it/resources/com/google/checkstyle/test/chapter5naming/rule53camelcase/InputCamelCaseDefined.java
 src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputSingleLineJavadocAndInvalidJavadocPosition.java
+src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnCanonicalConstructorsWithAnnotation.java
+src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnCompactConstructorsWithAnnotation.java
+src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnConstructorInRecord.java
+src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputRecordClassJavadocPosition.java
 src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectRequireEmptyLineBeforeBlockTagGroup.java
 src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule712paragraphs/InputIncorrectJavadocParagraph.java
 src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule713atclauses/InputJavaDocTagContinuationIndentation.java

--- a/src/it/java/com/google/checkstyle/test/chapter7javadoc/rule711generalform/GeneralFormTest.java
+++ b/src/it/java/com/google/checkstyle/test/chapter7javadoc/rule711generalform/GeneralFormTest.java
@@ -41,4 +41,48 @@ public class GeneralFormTest extends AbstractGoogleModuleTestSupport {
                 getPath("InputFormattedSingleLineJavadocAndInvalidJavadocPosition.java"));
     }
 
+    @Test
+    public void testJavadocPositionOnCompactConstructorInRecord() throws Exception {
+        verifyWithWholeConfig(getPath("InputJavadocPositionOnConstructorInRecord.java"));
+    }
+
+    @Test
+    public void testFormattedJavadocPositionOnCompactConstructorInRecord() throws Exception {
+        verifyWithWholeConfig(getPath("InputFormattedJavadocPositionOnConstructorInRecord.java"));
+    }
+
+    @Test
+    public void testJavadocPositionOnCompactConstructorWithAnnotation() throws Exception {
+        verifyWithWholeConfig(
+            getPath("InputJavadocPositionOnCompactConstructorsWithAnnotation.java"));
+    }
+
+    @Test
+    public void testFormattedJavadocPositionOnCompactConstructorWithAnnotation() throws Exception {
+        verifyWithWholeConfig(
+            getPath("InputFormattedJavadocPositionOnCompactConstructorsWithAnnotation.java"));
+    }
+
+    @Test
+    public void testJavadocPositionOnCanonicalConstructorWithAnnotation() throws Exception {
+        verifyWithWholeConfig(
+            getPath("InputJavadocPositionOnCanonicalConstructorsWithAnnotation.java"));
+    }
+
+    @Test
+    public void testFormattedJavadocPositionOnCanonicalConstructorWithAnnotation()
+            throws Exception {
+        verifyWithWholeConfig(
+            getPath("InputFormattedJavadocPositionOnCanonicalConstructorsWithAnnotation.java"));
+    }
+
+    @Test
+    public void testRecordClassJavadocPosition() throws Exception {
+        verifyWithWholeConfig(getPath("InputRecordClassJavadocPosition.java"));
+    }
+
+    @Test
+    public void testFormattedRecordClassJavadocPosition() throws Exception {
+        verifyWithWholeConfig(getPath("InputFormattedRecordClassJavadocPosition.java"));
+    }
 }

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedJavadocPositionOnCanonicalConstructorsWithAnnotation.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedJavadocPositionOnCanonicalConstructorsWithAnnotation.java
@@ -1,0 +1,86 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/** some javadoc. */
+public class InputFormattedJavadocPositionOnCanonicalConstructorsWithAnnotation {
+  @interface SizeType {
+    int max();
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   */
+  @SizeType(max = 3)
+  record Mapping(String from) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     */
+    @SizeType(max = 3)
+    Mapping(String from) {
+      this.from = from;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param to the param
+   */
+  @SizeType(max = 3)
+  public record Mapping2(String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param to The source
+     */
+    @SizeType(max = 3)
+    public Mapping2(String to) {
+      this.to = to;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to the to param
+   */
+  record BindCtor(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to The destination
+     */
+    BindCtor(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to the to param
+   */
+  public record BindCtor2(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to The destination
+     */
+    public BindCtor2(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedJavadocPositionOnCompactConstructorsWithAnnotation.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedJavadocPositionOnCompactConstructorsWithAnnotation.java
@@ -1,0 +1,58 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/** Some description. */
+public class InputFormattedJavadocPositionOnCompactConstructorsWithAnnotation {
+
+  @interface SizeType {
+    int max();
+  }
+
+  /**
+   * Something something.
+   *
+   * @param containerPath a path on the container
+   */
+  @SizeType(max = 2)
+  record BindMount(String containerPath) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param containerPath a path on the container
+     */
+    @SizeType(max = 1)
+    BindMount {}
+  }
+
+  /**
+   * Something something.
+   *
+   * @param goPath a path on the container
+   */
+  public record BindMount2(String goPath) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param goPath a path on the container
+     */
+    public BindMount2 {}
+  }
+
+  /**
+   * Something something.
+   *
+   * @param fileSystem a path on the container
+   */
+  @SizeType(max = 4)
+  public record MyRecord(String fileSystem) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param fileSystem a path on the container
+     */
+    @SizeType(max = 2)
+    public MyRecord {}
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedJavadocPositionOnConstructorInRecord.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedJavadocPositionOnConstructorInRecord.java
@@ -1,0 +1,78 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/** Some description. */
+public class InputFormattedJavadocPositionOnConstructorInRecord {
+
+  /**
+   * The configuration of a bind mount.
+   *
+   * @param containerPath a path on the container
+   * @param options mounting options
+   */
+  record BindMount(String containerPath, String... options) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param containerPath a path on the container
+     * @param options mounting options
+     * @throws NullPointerException if any of the arguments are null
+     */
+    BindMount {}
+  }
+
+  /**
+   * The configuration.
+   *
+   * @param text the text
+   */
+  public record MyRecord(String text) {
+
+    // violation 3 lines below 'Javadoc comment is placed in the wrong location.'
+    /** invalid comment. */
+    public MyRecord {}
+    /** invalid comment. */
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to the to param
+   */
+  record Mapping(String from, String to) {
+
+    // violation below 'Javadoc comment is placed in the wrong location.'
+    /** some javadoc. */
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to The destination
+     */
+    Mapping(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to the to param
+   */
+  public record Mapping2(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to The destination
+     */
+    public Mapping2(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedRecordClassJavadocPosition.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputFormattedRecordClassJavadocPosition.java
@@ -1,0 +1,71 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/**
+ * The configuration of a bind mount.
+ *
+ * @param containerPath a path on the container
+ * @param options mounting options
+ */
+public record InputFormattedRecordClassJavadocPosition(String containerPath, String... option) {
+
+  /**
+   * Creates a mount.
+   *
+   * @param containerPath a path on the container
+   * @param options mounting options
+   * @throws NullPointerException if any of the arguments are null
+   */
+  public InputFormattedRecordClassJavadocPosition {}
+
+  /**
+   * The configuration of a bind mount.
+   *
+   * @param containerPath a path on the container
+   */
+  record BindMount2(String containerPath) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param containerPath a path on the container
+     */
+    BindMount2 {}
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   */
+  record Mapping(String from) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     */
+    Mapping(String from) {
+      this.from = from;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to the to param
+   */
+  public record Mapping2(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to The destination
+     */
+    public Mapping2(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnCanonicalConstructorsWithAnnotation.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnCanonicalConstructorsWithAnnotation.java
@@ -1,0 +1,89 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/** some javadoc. */
+public class InputJavadocPositionOnCanonicalConstructorsWithAnnotation {
+
+  @interface SizeType {
+    int max();
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   */
+  @SizeType(max = 3)
+  record Mapping(String from) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     */
+    @SizeType(max = 3)
+    Mapping(String from) {
+      this.from = from;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param to the param
+   */
+  @SizeType(max = 3)
+  public record Mapping2(String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param to The source
+     */
+    @SizeType(max = 3)
+    public Mapping2(String to) {
+      this.to = to;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to   the to param
+   */
+  record BindCtor(String from, String to) {
+
+    // violation below 'Javadoc comment is placed in the wrong location.'
+    /** some javadoc. */
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to   The destination
+     */
+    BindCtor(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to   the to param
+   */
+  public record BindCtor2(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to   The destination
+     */
+    public BindCtor2(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnCompactConstructorsWithAnnotation.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnCompactConstructorsWithAnnotation.java
@@ -1,0 +1,60 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/** Some description. */
+public class InputJavadocPositionOnCompactConstructorsWithAnnotation {
+
+  @interface SizeType {
+    int max();
+  }
+
+  /**
+   * Something something.
+   *
+   * @param containerPath a path on the container
+   */
+  @SizeType(max = 2)
+  record BindMount(String containerPath) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param containerPath a path on the container
+     */
+    @SizeType(max = 1)
+    BindMount {}
+  }
+
+  /**
+   * Something something.
+   *
+   * @param goPath a path on the container
+   */
+  public record BindMount2(String goPath) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param goPath a path on the container
+     */
+    public BindMount2 {}
+  }
+
+  /**
+   * Something something.
+   *
+   * @param fileSystem a path on the container
+   */
+  @SizeType(max = 4)
+  public record MyRecord(String fileSystem) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param fileSystem a path on the container
+     */
+    @SizeType(max = 2)
+    public MyRecord {}
+    /** invalid javadoc.*/
+    // violation above 'Javadoc comment is placed in the wrong location.'
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnConstructorInRecord.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputJavadocPositionOnConstructorInRecord.java
@@ -1,0 +1,81 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/**
+ * Some description.
+ */
+public class InputJavadocPositionOnConstructorInRecord {
+
+  /**
+   * The configuration of a bind mount.
+   *
+   * @param containerPath a path on the container
+   * @param options       mounting options
+   */
+  record BindMount(String containerPath, String... options) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param containerPath a path on the container
+     * @param options       mounting options
+     * @throws NullPointerException     if any of the arguments are null
+     */
+    BindMount {
+    }
+  }
+
+  /**
+   * The configuration.
+   *
+   * @param text the text
+   */
+  public record MyRecord(String text) {
+
+    // violation 3 lines below 'Javadoc comment is placed in the wrong location.'
+    /** invalid comment. */
+    public MyRecord {}
+    /** invalid comment. */
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to   the to param
+   */
+  record Mapping(String from, String to) {
+
+    // violation below 'Javadoc comment is placed in the wrong location.'
+    /** some javadoc. */
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to   The destination
+     */
+    Mapping(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to   the to param
+   */
+  public record Mapping2(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to   The destination
+     */
+    public Mapping2(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+}

--- a/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputRecordClassJavadocPosition.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter7javadoc/rule711generalform/InputRecordClassJavadocPosition.java
@@ -1,0 +1,71 @@
+package com.google.checkstyle.test.chapter7javadoc.rule711generalform;
+
+/**
+ * The configuration of a bind mount.
+ *
+ * @param containerPath a path on the container
+ * @param options mounting options
+ */
+public record InputRecordClassJavadocPosition(String containerPath, String... option) {
+
+  /**
+   * Creates a mount.
+   *
+   * @param containerPath a path on the container
+   * @param options mounting options
+   * @throws NullPointerException if any of the arguments are null
+   */
+  public InputRecordClassJavadocPosition {}
+
+  /**
+   * The configuration of a bind mount.
+   *
+   * @param containerPath a path on the container
+   */
+  record BindMount2(String containerPath) {
+
+    /**
+     * Creates a mount.
+     *
+     * @param containerPath a path on the container
+     */
+    BindMount2 {}
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   */
+  record Mapping(String from) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     */
+    Mapping(String from) {
+      this.from = from;
+    }
+  }
+
+  /**
+   * Maps the ports.
+   *
+   * @param from the from param
+   * @param to the to param
+   */
+  public record Mapping2(String from, String to) {
+
+    /**
+     * The constructor for Mapping.
+     *
+     * @param from The source
+     * @param to The destination
+     */
+    public Mapping2(String from, String to) {
+      this.from = from;
+      this.to = to;
+    }
+  }
+}

--- a/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/utils/BlockCommentPosition.java
@@ -196,7 +196,8 @@ public final class BlockCommentPosition {
      * @return true if node is before compact constructor
      */
     public static boolean isOnCompactConstructor(DetailAST blockComment) {
-        return isOnTokenWithModifiers(blockComment, TokenTypes.COMPACT_CTOR_DEF)
+        return isOnPlainToken(blockComment, TokenTypes.COMPACT_CTOR_DEF, TokenTypes.IDENT)
+                || isOnTokenWithModifiers(blockComment, TokenTypes.COMPACT_CTOR_DEF)
                 || isOnTokenWithAnnotation(blockComment, TokenTypes.COMPACT_CTOR_DEF);
     }
 

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocPositionCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/InvalidJavadocPositionCheckTest.java
@@ -112,4 +112,26 @@ public class InvalidJavadocPositionCheckTest extends AbstractModuleTestSupport {
                 getPath("comment/package-info.java"), expected);
     }
 
+    @Test
+    public void testInvalidJavadocPositionOnCompactConstructors() throws Exception {
+        final String[] expected = {
+            "42:9: " + getCheckMessage(MSG_KEY),
+            "54:9: " + getCheckMessage(MSG_KEY),
+            "76:13: " + getCheckMessage(MSG_KEY),
+        };
+
+        verifyWithInlineConfigParser(
+            getPath("InputInvalidJavadocPositionOnCompactConstructors.java"), expected);
+    }
+
+    @Test
+    public void testInvalidJavadocPositionOnCompactConstructorsWithAnnotation() throws Exception {
+        final String[] expected = {
+            "47:9: " + getCheckMessage(MSG_KEY),
+        };
+
+        verifyWithInlineConfigParser(
+            getPath("InputInvalidJavadocPositionOnCompactConstructorsWithAnnotation.java"),
+                expected);
+    }
 }

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPositionOnCompactConstructors.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPositionOnCompactConstructors.java
@@ -1,0 +1,79 @@
+/*
+InvalidJavadocPosition
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.invalidjavadocposition;
+
+/**
+ * Some description.
+ */
+public class InputInvalidJavadocPositionOnCompactConstructors {
+
+    /**
+     * The configuration of a bind mount.
+     *
+     * @param containerPath a path on the container
+     * @param options       mounting options
+     */
+    record BindMount(String containerPath, String... options) {
+
+        /**
+         * Creates a mount.
+         *
+         * @param containerPath a path on the container
+         * @param options       mounting options
+         * @throws NullPointerException     if any of the arguments are null
+         */
+        BindMount {
+        }
+    }
+
+    /**
+     * The configuration.
+     *
+     * @param text the text
+     */
+    public record MyRecord(String text) {
+
+        // violation 2 lines below 'Javadoc comment is placed in the wrong location.'
+        public MyRecord {}
+        /** invalid comment. */
+    }
+
+    /**
+     * Maps the ports.
+     *
+     * @param from the from param
+     * @param to   the to param
+     */
+    record Mapping(String from, String to) {
+
+        // violation below 'Javadoc comment is placed in the wrong location.'
+        /** some javadoc. */
+        /**
+         * The constructor for Mapping.
+         *
+         * @param from The source
+         * @param to   The destination
+         */
+        Mapping(String from, String to) {
+            this.from = from;
+            this.to = to;
+        }
+    }
+
+    /**
+     * The configuration.
+     *
+     * @param text the text
+     */
+    public record MyRecord2(String text) {
+
+        // violation 2 lines below 'Javadoc comment is placed in the wrong location.'
+        public MyRecord2 {
+            /** invalid comment. */
+        }
+    }
+}

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPositionOnCompactConstructorsWithAnnotation.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPositionOnCompactConstructorsWithAnnotation.java
@@ -1,0 +1,88 @@
+/*
+InvalidJavadocPosition
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.invalidjavadocposition;
+
+public class InputInvalidJavadocPositionOnCompactConstructorsWithAnnotation {
+
+    @interface SizeType {
+        int max();
+    }
+
+    /**
+     * Something something.
+     *
+     * @param containerPath a path on the container
+     */
+    @SizeType(max = 2)
+    record BindMount(String containerPath) {
+
+        /**
+         * Creates a mount.
+         *
+         * @param containerPath a path on the container
+         */
+        @SizeType(max = 1)
+        BindMount {}
+    }
+
+    /**
+     * Something something.
+     *
+     * @param fileSystem a path on the container
+     */
+    @SizeType(max = 4)
+    public record MyRecord(String fileSystem) {
+
+        /**
+         * Creates a mount.
+         *
+         * @param fileSystem a path on the container
+         */
+        @SizeType(max = 2)
+        public MyRecord {}
+        /** some invalid javadoc */
+        // violation above 'Javadoc comment is placed in the wrong location.'
+    }
+
+    /**
+     * Maps the ports.
+     *
+     * @param from the from param
+     */
+    @SizeType(max = 3)
+    record Mapping(String from) {
+
+        /**
+         * The constructor for Mapping.
+         *
+         * @param from The source
+         */
+        @SizeType(max = 3)
+        Mapping(String from) {
+            this.from = from;
+        }
+    }
+
+    /**
+     * Maps the ports.
+     *
+     * @param to the param
+     */
+    @SizeType(max = 3)
+    public record Mapping2(String to) {
+
+        /**
+         * The constructor for Mapping.
+         *
+         * @param to The source
+         */
+        @SizeType(max = 3)
+        public Mapping2(String to) {
+            this.to = to;
+        }
+    }
+}

